### DISCUSSION TITLE
How to find launch with env value with AQL

### DISCUSSION THREAD

**Discussion by AlexPshe** [allure-framework/allure2#2374] [TestOps]
What I am trying to do: 
1. Filter Launches by Environment variable values
2. I need to find launches that have ENV var "qa version" starting from "241"

How can I do it with AQL?

**Comment by cheshi-mantu** [comment]
Hi @AlexPshe 
You can't directly do that as the values of env data are strings and AQL does not provide a feature allowing comparison of the values for the time being.
Could you please describe the global task, so we could think of an alternative?

cheers

PS please also consider using dedicated help desk for the Allure Testops users for similar inquiries (https://help.qameta.io)

**Comment by AlexPshe** [comment]
Hi @cheshi-mantu 

Well, the task I am trying to solve is: 
1. I have several launches of the same suites with different product versions which we provide in the env variable 
2. I want to build a dashboard with launches for different product versions
3. Now the dashboard includes all launches for all versions that does not make any sense to analyze together
<img width="817" alt="image" src="https://github.com/allure-framework/allure2/assets/17509584/63b17c59-fb64-4886-9aee-cb96f671aab0">

How can I solve this task?

**Reply by cheshi-mantu** [reply]
for your task there is no direct elegant solution I could think of

first that comes to mind, you can create something close to what is required by filtering launches by date, i.e. widget must show the data created after certain date 

<img width="511" alt="image" src="https://github.com/allure-framework/allure2/assets/8607348/071e51ce-470a-4a22-ab21-424761c8159e">

to filter our non-release launches, you need to have some distinguishable marker added either to name or as an env variable, and then exclude the launches that do not have this marker
<img width="720" alt="image" src="https://github.com/allure-framework/allure2/assets/8607348/6bd5689d-83a5-49d7-8192-7123198a6d45">
